Minor style consistency updates

sebastiangrimberg opened the pull request:
- Consistent indentation of doxygen docstrings as well as some missed formatting.
- Consistent file endings with newline separation `//------` in all `backends/` files. Previously this included a mix of no empty line preceeding the trailing comment, a newline, or no trailing comment.
- Consistent include orders according to https://libceed.org/en/latest/libCEEDdev/#include-what-you-use and corresponding change to `clang-format` settings.

jeremylt commented:
Re

>Consistent include orders in tests according to https://libceed.org/en/latest/libCEEDdev/#include-what-you-use and corresponding change to clang-format to not rearrange includes.

Personally, I'm happy letting `clang-format` sort that for us. Then the convention becomes "use `clang-format`" so its automatically done for us.

sebastiangrimberg commented on interface/ceed-preconditioning.c:
```diff
@@ -1126,13 +1127,12 @@ int CeedOperatorGetOperatorAssemblyData(CeedOperator op, CeedOperatorAssemblyDat
+  For each active CeedBasis, the CeedOperatorAssemblyData holds an array of all input and output CeedEvalModes for this CeedBasis.
+  The CeedOperatorAssemblyData holds an array of offsets for indexing into the assembled CeedQFunction arrays to the row representing each
+CeedEvalMode. The number of input columns across all active bases for the assembled CeedQFunction is also stored. Lastly, the CeedOperatorAssembly
+data holds assembled matrices representing the full action of the CeedBasis for all CeedEvalModes.
```
Good idea and I've gone through in 988944f to try to break sentences onto their own lines where I could find them. I know there is a `ReflowComments` parameter for `clang-format` to disable wrapping of comments, but unfortunately it would also do it for non-docstrings where this can be quite useful I think. So maybe for now it makes sense to keep doing it manually.

sebastiangrimberg commented:
> Personally, I'm happy letting `clang-format` sort that for us. Then the convention becomes "use `clang-format`" so its automatically done for us.

Indeed this was my mistake for the files in `tests/` and I reverted that change. However I did add [c85e864](https://github.com/CEED/libCEED/pull/1197/commits/c85e864000754611328ed43fe87e29f99787d593) which will at least enforce the priorities as desired more than just alphabetically. Note that in order to avoid a lot of whitespace changes I kept `IncludeBlocks: Preserve` for now, which means the sorting only occurs in contiguous blocks rather than merging all the `#include` statements and sorting.

jeremylt commented:
For context, the guidelines in https://libceed.org/en/latest/libCEEDdev/#include-what-you-use predate our use of `clang-format` and it didn't get updated when we switched. (`astyle` didn't reorder includes). I'm fine keeping your new rules in `.clang-format` to enforce what we used to do, but I'm equally happy using the `clang-format` defaults and removing that last paragraph of that section.

sebastiangrimberg commented:
> I'm fine keeping your new rules in .clang-format to enforce what we used to do, but I'm equally happy using the clang-format defaults and removing that last paragraph of that section.

Cool, thanks. From my side I tend to agree with the old guidelines of keeping `ceed` headers first, so if we can automate it with `clang-format` and the new rules I think we might as well do that. I'll mark this as "in review" now and stop any further changes other than addressing review comments.

jeremylt reviewed (approved):
LGTM
